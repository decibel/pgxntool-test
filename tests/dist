#!/bin/bash

trap 'echo "ERROR: $BASH_SOURCE: line $LINENO" >&2' ERR
set -o errexit -o errtrace -o pipefail
#set -o xtrace -o verbose

if [ "$1" == "-v" ]; then
  verboseout=1
  shift
fi
BASEDIR=`cd ${0%/*}; pwd`

. $BASEDIR/../.env
. $TOPDIR/lib.sh
cd $TEST_REPO

DISTRIBUTION_NAME=distribution_test

# Note: It's easier to do this now than when the checkout is all cluttered
out Test creating a release
make dist
unzip -l ../$DISTRIBUTION_NAME-0.1.0.zip | grep .asc | awk '{print $4}'
# grep exits with 1 if it can't find anything
out "Ensure there's at least some docs in the distribution" # This is mostly to make sure the next test works
unzip -l ../$DISTRIBUTION_NAME-0.1.0.zip | egrep -q 'asc|adoc|asciidoc|html|md|txt' || exit 1
out Ensure there are no pgxntool docs in the distribution
if unzip -l ../$DISTRIBUTION_NAME-0.1.0.zip | grep pgxntool/ | egrep 'asc|adoc|asciidoc|html|md|txt'; then
  die 5 "Found document files in pgxntool/"
fi

check_log

# vi: expandtab sw=2 ts=2
