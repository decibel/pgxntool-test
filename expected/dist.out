# Test creating a release
git branch 0.1.0
git push --set-upstream origin 0.1.0
To ../fake_repo
 * [new branch]      0.1.0 -> 0.1.0
Branch 0.1.0 set up to track remote branch 0.1.0 from origin.
git archive --prefix=distribution_test-0.1.0/ -o ../distribution_test-0.1.0.zip 0.1.0
distribution_test-0.1.0/t/TEST_DOC.asc
distribution_test-0.1.0/t/doc/asc_doc.asc
distribution_test-0.1.0/t/doc/asciidoc_doc.asciidoc
# Ensure there's at least some docs in the distribution
# Ensure there are no pgxntool docs in the distribution
